src/Mod/Sketcher/App/planegcs/Geo.cpp

```diff
@@ -773,6 +773,189 @@ DeriVector2 BSpline::CalculateNormal(const Point& p, const double* derivparam) c
     return ret;
 }
 
+DeriVector2 BSpline::CalculateNormal(const double* param, const double* derivparam) const
```

Comment:
This method has grown long enough that I'd suggest refactoring it immediately, rather than doing that as a separate PR later.

Reply:
Sorry I might have forgotten: have we discussed refactoring this?

Reply:
No, I don't think so. But with a cognitive complexity of 48 it's definitely needed.

Reply:
What is that? And how do you suggest it can be reduced?